sql/in_mem_table/inmem_table_data.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package in_mem_table
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Eventually these structures should be replaced by memory tables, however at the time of writing this file, the
```

Comment:
If this is actually necessary (I don't believe it is, see the overall PR comments) then we should just bite the bullet and make memory tables fast / work with indexes, it's past time